clippy2021.md

```diff
@@ -0,0 +1,178 @@
+# Clippy Roadmap 2021
+
+As the Rust language gets bigger and bigger, also Clippy gets more and more
+attention. This is good for the project, as it gets more and more users and
+contributors. With that we are seeing more and more issues about the reliability
+and usability of Clippy. Some examples are:
+
+- `cargo clippy` after `cargo check` is the most
+  prominent and has many duplicate issues referring to it:
+  [#4612](https://github.com/rust-lang/rust-clippy/issues/4612)
+- Better testing of clippy lints before enabling them by default:
+  [#6429](https://github.com/rust-lang/rust-clippy/issues/6429)
+- Add an easy way to run clippy with high FPR lints disabled:
+  [#5537](https://github.com/rust-lang/rust-clippy/issues/5537)
+- Missing warn-by-default correctness group:
+  [#6366](https://github.com/rust-lang/rust-clippy/issues/6366)
+
+And many more, that are either duplicates or address other things about
+reliability and usability.
+
+Besides that, there are internal things that aren't optimal currently. Those
+span from Clippy development to management of Clippy. This document will list
+things, we should address in 2021 and should start a conversation how we should
+prioritize things. It is split up into external and internal issues, which are
+further split up.
+
+## External
```

Comment:
Some other user facing stuff I would like to improve:
* Stabilizing `--fix`, and going through the lint list to see if we can turn more lints into `MachineApplicable`
* Support for multi-span suggestions in rustfix.

Reply:
Stabilizing fix goes with the `cargo check -> cargo clippy` stabilization, I think? Multi-span suggestions is out of scope for this document, because it is a rustfix issue.

Reply:
> Stabilizing fix goes with the `cargo check -> cargo clippy` stabilization, I think? 

Yes, you're right.

>Multi-span suggestions is out of scope for this document, because it is a rustfix issue.

Although it's a rustfix issue, I think we are the more interested party in getting that to work.